spec/flow.md

```diff
@@ -0,0 +1,37 @@
+# Flow
+
+This document describes the contract flows grouped by how they will be called by clients. 
+
+- How to read the images?
+
+Each flow transaction starts from a user or liquidation bot. The farthest left is the first transaction followed by the next one in the row. 
+
+## Create Account
+
+A user creates an account and deposits funds into the obligation account. 
+
+![Create Account](./assets/create_acc.png)
+
+## Open
+
+Open starts with a user setting a leverage level, borrows the funds needed, then creates a new order. At this stage the account is marked as having an open trade, prohibiting an withdrawals until the loan has been repaid in full with interest. 
+
+![Open](./assets/open.png)
+
+## Close
+
+Close begins with first creating a new order on the serum dex to close the open trade.
+
+![Close](./assets/close.png)
+
+## Withdraw
+
+Withdraw settles the funds by calling the settle function on the serum dex. Once it has received the funds it will repay the loan. In the process of repaying the loan, if in the profit, the obligation will be sent to the user.
```

Comment:
When you say sends funds to program account do you mean the margin account

Reply:
Also isn't it the `settle` call first made through the margin account which then calls `settle` on the serum DEX

Reply:
We may need to also sell some of the funds from the obligation account